### ISSUE TITLE
CTS profile does not match

### ISSUE THREAD

**Issue by j0110** [kdrag0n/safetynet-fix#250]
Hi
I just switched to LOS 20, and CTS doesn't pass anymore
![Screenshot_20230124-121507_Yet Another SafetyNet Attestation Checker](https://user-images.githubusercontent.com/23175256/214277875-9e5db416-c518-4a34-a678-2eb1ae14f593.png)
I got USNF, samiko, and it used to work with LOS19.1
Any ideas?
Thanks a lot

**Comment by j0110** [comment]
Surprisingly it works with https://github.com/Displax/safetynet-fix/releases/tag/v2.3.1-MOD_2.1...

**Comment by Displax** [comment]
> Surprisingly it works with https://github.com/Displax/safetynet-fix/releases/tag/v2.3.1-MOD_2.1...

Because of microG. Use official GMS.

**Comment by ibart** [comment]
> Surprisingly it works with https://github.com/Displax/safetynet-fix/releases/tag/v2.3.1-MOD_2.1...

Indeed. I rolled back to that. But with 2.4 if you retry the test for a while you could pass the test.

**Comment by j0110** [comment]
Same as #248

**Comment by stanelie** [comment]
I've noticed that just switching on the deny list in Magisk and switching it back off is enough to make the issue go away for a time.

Maybe it's unrelated...

**Comment by Djtrip83** [comment]
Just adding my feedback as I have also been affected since this morning. Issue is only affecting Google Pay but it is constant (no on and off like reported on [#248](https://github.com/kdrag0n/safetynet-fix/issues/248)). All other apps cannot detect root and banking apps work fine. At one point shortly after boot I noticed YASNAC passed all checks and then didn't when I re-ran a few seconds later. Things looks fine at boot but after boot something is changed maybe?

**Model:** Pixel 6a bluejay 

**Magisk:** 25.2 stable hidden with all Google apps (incl. system) on DenyList as per usual

**Magisk Modules:**
- MHPG 6.1.2-v137 with spoof fingerprint Pixel 2 and force basic attestation and now removed both altogether
- Shamiko 0.6 (Enforce Denylist therefore not ticked off 
- USNF 2.3.1 and now updated to 2.4.0

**Additional notes:**
- YASNAC passes Basic integrity but fails CTS Profile
- Play Integrity API Checker meets Basic integrity but fails Device integrity and fails Meets Strong Integrity
- PlayStore states Device is not certified

**Comment by Mitezuss** [comment]
I have the same problem.
Tried all the possible fix mentioned.

Too has been working fine until yesterday.

So, maybe some update ?
I have the problem on GSI ROM only. Work fine on STOCK or OEM STOCK PORTED

**Comment by LambdaSoft** [comment]
Maybe this is the solution (and the problem): https://github.com/hentaiOS/platform_frameworks_base/commit/32cd021753626ae2dc0f72d222ac4fd734f3a8aa 
It seems that now Google checks the product and model names against the fingerprint

**Comment by Jnss98b** [comment]
I just noticed the same problem today too on my pixel 7 pro. Most safetynet/play integrity enabled apps work fine but google pay refuses to let me use tap to pay, yasnac returns basic integrity passed with cts profile fail, and play integrity checker returns true for MEETS_BASIC_INTEGRITY but false for the other 2

**Comment by Djtrip83** [comment]
> Maybe this is the solution (and the problem): https://github.com/hentaiOS/platform_frameworks_base/commit/32cd021753626ae2dc0f72d222ac4fd734f3a8aa 
> It seems that now Google checks the product and model names against the fingerprint

Ah! I noticed yesterday that my spoofed fingerprint was read as Pixel 2 Bluejay instead of Pixel 2 walleye. I then checked props and most were not changed by USNF and MHPC (only model was). I was about to change them all manually but would take too long...

Could this be the culprit?

**Comment by ShiroBlank** [comment]
> Surprisingly it works with https://github.com/Displax/safetynet-fix/releases/tag/v2.3.1-MOD_2.1...

Big vouch for this, works perfectly, just installed this instead of the stock USNF and boom, works.

**Comment by Mak3yrM0v3** [comment]
OnePLus Nord AC2003 here whit PixelExperience13 cant bypass even whit Displax mod. CTS profile mismatches.

**Comment by antonywu** [comment]
> > Surprisingly it works with https://github.com/Displax/safetynet-fix/releases/tag/v2.3.1-MOD_2.1...
> 
> Big vouch for this, works perfectly, just installed this instead of the stock USNF and boom, works.

I did not use that one, but opt for a more recent release https://github.com/Displax/safetynet-fix/releases/download/v2.4.0-MOD_1.2/safetynet-fix-v2.4.0-MOD_1.2.zip

Verified to work with Pixel 6a (13), and OnePlus 8 (LOS 20). Google Wallet tap to pay works again.

**Comment by XDM-Inc** [comment]
> Surprisingly it works with https://github.com/Displax/safetynet-fix/releases/tag/v2.3.1-MOD_2.1...

wow,  i have been struggling to get miui to pass with all modern methods and this one fixed it! 👍

**Comment by adipascu** [comment]
The fix is in #261

**Comment by gio1135** [comment]
> The fix is in #261

what am i supposed to do with that

**Comment by Displax** [comment]
> > The fix is in #261
> 
> what am i supposed to do with that

https://github.com/Displax/safetynet-fix/releases/tag/v2.4.0-MOD_1.2

**Comment by Sandalu123** [comment]
> Surprisingly it works with https://github.com/Displax/safetynet-fix/releases/tag/v2.3.1-MOD_2.1...

That's the rocket to the mars. Thanks
